### PR TITLE
Add go/bump pipeline and convert 5 packages to it.

### PR DESCRIPTION
**Pull request by dlorenc**
Fixes:

Related:

### CODE REVIEW

File: pipelines/go/bump.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Bump go deps to a certain version
+
+# Leave go off here, some packages pin to an older version. Let's just assume this gets injected elsewhere.
```
Comment (mattmoor) [inline pipelines/go/bump.yaml:3]: You could check this by running `which go` or something in the pipeline?

File: caddy.yaml [hunk 2]
```diff
@@ -25,13 +25,16 @@ pipeline:
       install -m644 -D "./Caddyfile" "${{targets.destdir}}/etc/caddy/Caddyfile"
       install -m755 -D "./index.html" "${{targets.destdir}}/usr/share/caddy/index.html"
 
+  - uses: go/bump
+    with:
+      deps: google.golang.org/grpc@v1.58.3 go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.44.0 go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc@v1.19.0
```
Comment (mattmoor) [inline caddy.yaml:30]: Hmm, this is a different set of things than the LHS.  I'm curious what motivated the difference?

File: boring-registry.yaml [hunk 3]
```diff
@@ -23,6 +23,11 @@ pipeline:
       expected-commit: 083aa721311a3745ee0e5bd3b68bfce7ed2c1f4c
       tag: v${{package.version}}
 
+  - uses: go/bump
+    with:
+      # CVE-2023-39325 and CVE-2023-3978
+      deps: golang.org/x/net@v0.17.0 google.golang.org/grpc@v1.56.3
```
Comment (mattmoor) [inline boring-registry.yaml:29]: same question re: diffs

File: caddy.yaml (hunk 2 above)
Comment (dlorenc) [reply caddy.yaml:30]: The grpc one flagged in the scanning ci check.

File: boring-registry.yaml (hunk 3 above)
Comment (dlorenc) [reply boring-registry.yaml:29]: The grpc one flagged at CI time so I manually added it. We need advisories for these.

**Review by mattmoor — APPROVED** [review]

**Pull request merged**